modules/media/ajax/FileUpload.php

```diff
@@ -390,3 +396,27 @@ function getFilesList()
 
     return $mediaFiles;
 }
+
+/**
+ * Checks that a date is not in the future.
+ *
+ * @param string $dateTaken The date (in YYYY-MM-DD
+ *                          format) to check)
+ *
+ * @return void (but may terminate as a side-effect)
+ */
+function checkDateTaken($dateTaken)
+{
+    if (!empty($dateTaken)) {
+        $date = date_create_from_format("Y-m-d", $dateTaken);
+        if ($date === false) {
+            showMediaError("Invalid date: $dateTaken", 400);
+        }
+
+        $now  = new DateTime();
+        $diff = date_diff($date, $now)->format("%a");
+        if ($diff > 0) {
```

Comment:
@driusan Shouldn't this be `if ($diff < 0)` ??
`date_diff` does `$now - $date` which should technically be positive if the `$date` is in the past...